framework/include/controls/ControlInterface.h

```diff
@@ -134,93 +91,54 @@ class ControlInterface
 {
-  // Get the InputParameter objects that contain objects
-  ControlParameterName desired = tokenizeName(name);
-  std::vector<InputParameterIterator> iters = getValidInputParameterIteratorsByName<T>(name, desired);
-
-  // If more than one parameter is located, report an error message listing the object and parameter names
-  if (iters.size() > 1)
-  {
-    std::ostringstream oss;
-    oss << "The controlled parameter, '" << name << "', in " << _ci_name << " was found in multiple objects:\n";
-    for (std::vector<InputParameterIterator>::const_iterator it = iters.begin(); it != iters.end(); ++it)
-      oss << "  " << (*it)->first << '\n';
-    oss << "\nUse the method 'getControlParamVector' to access multiple parameters";
-    mooseError(oss.str());
-  }
-
-  // Return a writeable reference to the parameter
-  return iters[0]->second->set<T>(desired.param);
+  return getControlParamByName<T>(_ci_parameters.get<std::string>(name), count);
 }
 
 template<typename T>
-std::vector<T*>
-ControlInterface::getControlParamVector(const std::string & name)
+ControllableParameter<T>
+ControlInterface::getControlParamByName(const std::string & name)
 {
-  return getControlParamVectorByName<T>(_ci_parameters.get<std::string>(name));
+  return getControlParamByName<T>(_ci_parameters.get<std::string>(name),
+                                  std::numeric_limits<unsigned int>::max());
 }
 
 template<typename T>
-std::vector<T*>
-ControlInterface::getControlParamVectorByName(const std::string & name)
+ControllableParameter<T>
+ControlInterface::getControlParamByName(const std::string & name, unsigned int count)
 {
-  // Get the InputParameter objects that contain objects
-  ControlParameterName desired = tokenizeName(name);
-  std::vector<InputParameterIterator> iters = getValidInputParameterIteratorsByName<T>(name, desired);
-
-  // The vector the return
-  std::vector<T*> output;
-
-  // Loop over all InputParameter objects
-  for (std::vector<InputParameterIterator>::iterator it = iters.begin(); it != iters.end(); ++it)
-    output.push_back(&((*it)->second->set<T>(desired.param)));
-
-  return output;
-}
 
-template<typename T>
-std::vector<InputParameterIterator>
-ControlInterface::getValidInputParameterIteratorsByName(const std::string & name, const ControlParameterName & desired)
-{
-  // The vector to return
-  std::vector<InputParameterIterator> output;
+  // The desired parameter name
+  MooseObjectParameterName desired(name);
 
-  // Storage for tokenizing the current InputParameters name
-  std::vector<std::string> current;
+  // The ControlllableParametr object to return
+  ControllableParameter<T> output;
 
   // Loop over all InputParameter objects
-  for (InputParameterIterator it = _input_parameter_warehouse.begin(_tid); it != _input_parameter_warehouse.end(_tid); ++it)
+  for (InputParameterWarehouse::InputParameterIterator it = _input_parameter_warehouse.begin(_tid); it != _input_parameter_warehouse.end(_tid); ++it)
```

Comment:
I would be happy to get rid of this typedef once we nail down the implementation, the type has changed a lot in my iterations and this makes that much easier.